HousingFinanceInterimApi/V1/Gateways/UPCashLoadGateway.cs

```diff
@@ -33,7 +35,66 @@ public async Task LoadCashFiles()
         {
             try
             {
-                await _context.LoadCashFiles().ConfigureAwait(false);
+                // Clear cash dumps for filenames that failed to process
+                var failedFileNameIds = _context.UpCashDumpFileNames
+                    .Where(fileName => !fileName.IsSuccess)
+                    .Select(fileName => fileName.Id)
+                    .ToList();
+                _context.UpCashDumps
+                    .Where(cashDump => failedFileNameIds.Contains(cashDump.UPCashDumpFileNameId))
+                    .BatchDelete();
+
+                var unreadCashDumps = _context.UpCashDumps.Where(cashDump => !cashDump.IsRead).ToList();
+
+                decimal parseAmountPaid(string amountPaidSubstring)
+                {
+                    // Example: +0000000001.00
+                    bool isPositive = amountPaidSubstring[0] == '+';
+                    var amountString = amountPaidSubstring[1..];
+                    var amount = Math.Round(decimal.Parse(amountString), 2);
+                    if (!isPositive)
+                        amount *= -1;
+                    return amount;
+                }
```

Comment:
I think this is very specific to this procedure because it's the only one that parses this string format as far as I know. I would have written this inline if it was possible in dotnet, but failing that I at least wanted the code to be very close to where it's used so it's easy to reference. I could make it a private method but there's a tradeoff with that requiring the reader to click into it.

Reply:
Turns out C# parses the + or - already in decimal.tryparse so the method could be removed

https://github.com/LBHackney-IT/housing-finance-interim-api/pull/149/commits/4f55cca44a46d72c8cea99ff4c2e92e6c7d92f1f